icu4j/main/core/src/main/java/com/ibm/icu/util/MeasureUnit.java

```diff
@@ -496,8 +502,14 @@ public String getIdentifier() {
      * @stable ICU 68
      */
     public Complexity getComplexity() {
+        if (isComplexitySet) {
+            return complexity;
+        }
```

Comment:
This is not thread-safe. Please read https://en.wikipedia.org/wiki/Double-checked_locking